modules/actions/snapshots/lib/apply-usd-values.ts

```diff
@@ -0,0 +1,88 @@
+import { Prisma, Chain } from '@prisma/client';
+import { prisma } from '../../../../prisma/prisma-client';
+import { now, roundToMidnight } from '../../../common/time';
+import _ from 'lodash';
+
+export const applyUSDValues = async (
+    rawSnapshots: Prisma.PrismaPoolSnapshotUncheckedCreateInput[],
+    fetchPrices: (chain: Chain, timestamp?: number) => Promise<Record<string, number>> = fetchPricesHelper,
+    fetchPoolTokens: (poolIds: string[]) => Promise<Record<string, any[]>> = fetchPoolTokensHelper,
+): Promise<Prisma.PrismaPoolSnapshotUncheckedCreateInput[]> => {
+    const lastMidnight = roundToMidnight(now());
```

Comment:
think this is incorrect. It actually gives a different timestamp on 11am than it does on 1pm. Should be the same though.